src/guides/components/sqlite.md

```diff
@@ -0,0 +1,89 @@
+# Storing Data in SQLite
+
+For storing data generated by your plugin, Caido utilizes SQLite databases.
+
+SQLite is a lightweight database engine made available via a small library. It requires no setup, administration or separate server. Instead, all data is stored in a single file.
+
+The `sqlite` interface can be used to create and interact with a SQLite database file to store your Caido plugin data. Operations will be asynchronous using a connection pool of worker threads.
+
+The valid parameter data types that can be stored are: `null`, `number`, `bigint`, `string`, and `Uint8Array`.
+
+## SDK Database Operations
+
+The SDK offers the following operations that can be used on the database object:
+
+The `result` represents the outcome of the database operation:
+
+- `changes`: Returns the number of rows affected by the operation.
+- `lastInsertRowid`: Returns the row ID of the last inserted row.
+- `.open()`: Returns the database.
+- `.prepare()`: Compiles a SQL statement for future execution.
+- `.exec()`: Executes one or more statements without returning any results.
+- `.get()`: Executes a prepared statement and returns the first result as an object.
+- `.run()`: Executes a prepared statement and returns an object of the resulting changes.
+
+The connection pooling can be configured using `OpenOptions`:
+
+- `maxConnections`: The maximum number of concurrent connections. By default it uses 5.
+- `minConnections`: The minimum number of concurrent connections to maintain. By default it is set to 0.
+- `idleTimeout`: How long connections can remain idle.
+- `maxLifetime`: The maximum lifespan of connections.
```

Comment:
This is reference material. We should redirect the reader to the appropriate reference page in https://developer.caido.io/reference/sdks/backend/ instead of listing them here.